Quaver.Shared/Screens/Result/UI/ResultScoreContainer.cs

```diff
@@ -270,6 +280,125 @@ private void CreateKeyValueItems()
+            var largestHitWindow = processor.JudgementWindow.Values.Max();
+
+            double mean = 0.0, standardDeviation = 0.0;
+            if (processor.Stats != null)
+            {
+                var hitDifferences = new List<int>();
+                var sum = 0;
+
+                foreach (var breakdown in processor.Stats)
+                {
+                    var hitDifference = breakdown.HitDifference;
+
+                    // The LN releases are _not_ scaled here because we want an accurate mean.
+
+                    // No need to check for Type == Miss as all of them have hitDifference == int.MinValue.
+                    if (hitDifference != int.MinValue && Math.Abs(hitDifference) <= largestHitWindow)
+                    {
+                        hitDifferences.Add(hitDifference);
+                        sum += hitDifference; // This overflows at like 13 million max judgements.
+                    }
+                }
+
+                var count = hitDifferences.Count();
+                if (count > 0)
+                {
+                    mean = (double) sum / (double) count;
+                    standardDeviation = Math.Sqrt(hitDifferences.Average(v => Math.Pow(v - mean, 2)));
+
+                    // Undo the rate scaling.
+                    mean /= ModHelper.GetRateFromMods(processor.Mods);
+                    // Since variance(ax) = a^2 variance(x), then std(ax) = a std(x)
+                    standardDeviation /= ModHelper.GetRateFromMods(processor.Mods);
+                }
+            }
+
+            var availableWidth = Width - VerticalDividerLine.X;
+            var buttonPadding = 15;
+            var buttonWidth = (availableWidth - buttonPadding * 3) / 2;
+
+            var localOffsetButton = new BorderedTextButton("Adjust Local Offset", Colors.MainAccent,
+                (o, e) =>
+                {
+                    // Local offset is scaled with rate, so the adjustment depends on the rate the
+                    // score was played on.
+                    var change = mean * ModHelper.GetRateFromMods(processor.Mods);
+                    ResultScreen.Map.LocalOffset =
+                        (int) Math.Round(ResultScreen.Map.LocalOffset - change);
+
+                    MapDatabaseCache.UpdateMap(ResultScreen.Map);
+                    NotificationManager.Show(NotificationLevel.Success,
+                        $"Local offset was set to {ResultScreen.Map.LocalOffset} ms.");
+                })
+            {
+                Parent = this,
+                X = VerticalDividerLine.X + buttonPadding,
+                Y = BottomHorizontalDividerLine.Y - 15 - 25,
+                Height = 25,
+                Width = buttonWidth,
+                Text =
+                {
+                    FontSize = 9
+                }
+            };
+            var globalOffsetButton = new BorderedTextButton("Adjust Global Offset", Colors.MainAccent,
```

Comment:
The one issue I have with these buttons are that there should probably be some sort of confirmation button making the user aware of what their new offset will be and if they'd like to proceed. It can be in the form of a `ConfirmCancelDialog`. If not this, then the option to undo should be there, but I prefer the former.

Reply:
Also, it's probably only worth creating these buttons only after gameplay. You wouldn't want to always display it (especially when clicking on an online leaderboard score or watching a replay)

Reply:
There is value in having them on local scores: I used them a couple of times to adjust the offset from a score that I had before this was implemented, and also a couple of times when I forgot to click the button before escaping out of the gameplay score.

Reply:
There are some instances where you don't have access to these stats such as online scores and multiplayer (for other players). Hiding it when `Score.IsOnline` is true should do the trick.